src/test/java/seedu/dengue/logic/parser/DeleteCommandParserTest.java

```diff
@@ -42,4 +68,79 @@ public void parse_validArgsMultiIndex_returnsDeleteCommand() {
     public void parse_invalidArgsMultiIndex_throwsParseException() {
         assertParseFailure(parser, "1 2 a", String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE));
     }
+
+    // date
+
+    @Test
+    public void parse_validArgsDate_returnsDeleteCommand() {
+        DeleteDatePredicate predicate = new DeleteDatePredicate(Optional.of(new Date(VALID_DATE_ALICE)));
+        assertParseSuccess(parser, " d/2023-03-05", new DeleteCommand(predicate));
+    }
+
+    // range
+
+    @Test
+    public void parse_validArgsCompleteRange_returnsDeleteCommand() {
+
+        Range<Date> range = ContinuousData.generateRange(
+                new StartDate(Optional.of(new Date(VALID_DATE_BENSON))),
+                new EndDate(Optional.of(new Date(VALID_DATE_ALICE))));
+        DeleteDatePredicate predicate = new DeleteDatePredicate(range);
+
+        assertParseSuccess(parser, " sd/2022-03-05 ed/2023-03-05", new DeleteCommand(predicate));
+    }
+
+    //    @Test
+    //    public void parse_validArgsPartialRangeStartGiven_returnsDeleteCommand() {
+    //
+    //        Range<Date> range = ContinuousData.generateRange(
+    //                new StartDate(Optional.of(new Date(VALID_DATE_ALICE))),
+    //                new EndDate(Optional.empty()));
+    //        DeleteDatePredicate predicate = new DeleteDatePredicate(range);
+    //
+    //        assertParseSuccess(parser, " sd/2022-03-05", new DeleteCommand(predicate));
+    //    }
+    //
+    //    @Test
+    //    public void parse_validArgsPartialRangeEndGiven_returnsDeleteCommand() {
+    //
+    //        Range<Date> range = ContinuousData.generateRange(
+    //                new StartDate(Optional.empty()),
+    //                new EndDate(Optional.of(new Date(VALID_DATE_ALICE))));
+    //        DeleteDatePredicate predicate = new DeleteDatePredicate(range);
+    //
+    //        assertParseSuccess(parser, " ed/2022-03-05", new DeleteCommand(predicate));
+    //    }
```

Comment:
delete this if not in use?

Reply:
It's commented out because the test is failing for some reason, even though it seems to work fine on the app! Would like to keep it there as a reference to come back and work with.

Reply:
Fixed! I was tired last night and didn't realise I'd written the wrong date lol.